src/actions/player/index.js

```diff
@@ -90,12 +97,31 @@ function getIteratedTrack(playlist, currentActiveTrackId, iterate) {
   return playlist[index + iterate];
 }
 
+function getRandomTrack(playlist, currentActiveTrackId) {
+  const index = findIndex(isSameTrack(currentActiveTrackId), playlist);
+  function getRandomIndex() {
+    const randNum = Math.floor(Math.random() * playlist.length);
+    if (randNum === index) {
```

Comment:
That is the only issue I see. When you have only one track in your playlist, this will end up in an endless recursive loop. You could prevent it by adding either `if (randNum === index && playlist.length > 1) { ... }` or by not showing the shuffle button in the first place when the playlist isn't longer than 1 track. If you stick with the latter, you would also have to prevent the shuffle mode (which could be enabled) even when the button is hidden.